week_2/prime_numbers.rb

```diff
@@ -14,5 +14,29 @@
 # We will raise an `ArgumentError` exception to let the caller know that
 # their function arguments were incorrect.
 def prime_numbers(n:)
-  raise NotImplementedError # TODO
+  if n < 0
+    raise ArgumentError.new("Negative Input is not allowed!")
+  end
+  
+  ans = Array.new()
+  index = 2
+  index2 = 2
+  flag = true
+  while index <= n
```

Comment:
`each` or `for` loop can also be used. [Ruby ranges](https://www.rubyguides.com/2016/06/ruby-ranges-how-do-they-work/)

```ruby
(2..n).each do |num|
  # ...
end
```

Reply:
Thanks. will look into them too